src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -804,7 +813,9 @@ intptr_t ObjectSynchronizer::FastHashCode(Thread* current, oop obj) {
     markWord temp, test;
     intptr_t hash;
     markWord mark = read_stable_mark(obj);
-
+    if (UseHeavyMonitors && VerifyHeavyMonitors) {
```

Comment:
VerifyHeavyMonitors should require that UseHeavyMonitors be set. There should be logic in arguments.cpp to check that.